### DISCUSSION TITLE
[WINUI Only] MediaPicker.PickPhotoAsync throws System.UnauthorizedAccessException

### DISCUSSION THREAD

**Discussion by enkaradag** [CommunityToolkit/Maui#1866] [Q&A]
Hi,

I have an option in my app to choose an image file from disk (that opens old OpenFileDialog)

I am using MediaPicker.Default.PickPhotoAsync

I set my app as EntryPoint="Windows.PartialTrustApplication" in Package.appxmanifest, so MediaPicker throws System.UnauthorizedAccessException: 'Attempted to perform an unauthorized operation.' (Seems my app do not have access to storage)

(Microsoft Store does not like runFullTrust apps, thats why is set it like that)

If i enable "runFullTrust", PickPhotoAsync runs as expected. But i dont want to enable "runFullTrust". I added removableStorage, videosLibrary, picturesLibrary but those didn't help.

How can i add only this Capability (open the file open dialog) to my WinUI app?

Here is capabilities section of my Package.appxmanifest file

```
  <Capabilities>
    <!--<rescap:Capability Name="runFullTrust"/>-->
    <uap6:Capability Name="graphicsCapture"/>
    <uap:Capability Name="removableStorage"/>
    <uap:Capability Name="videosLibrary"/>
    <uap:Capability Name="picturesLibrary"/>
    <Capability Name="internetClient"/>
  </Capabilities>
```

and here is the Application section

```
<Applications>
    <Application Id="App" Executable="$targetnametoken$.exe" EntryPoint="Windows.PartialTrustApplication">
    ...
</Application>
```

My toast messages (Toast.Make) are also stuck with UnauthorizedAccessException. I couldn't figure out how to add that capability too.

That took my whole day. Any help will be appreciated

Regards

Ender

**Comment by bijington** [comment]
It has been suggested by a bug report on the documentation repository that you might need to follow the same steps for Toast as we require for SnackBar: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/alerts/snackbar?tabs=windows%2Cios

**Reply by enkaradag** [reply]
Thank you.

windows.toastNotificationActivation seems to require runFullTrust. That example is also created with runFullTrust scenario.

**Comment by VladislavAntonyuk** [comment]
Also Microsoft store works fine with runFullTrust if you mention you are using MAUI. The app is approved.

**Reply by enkaradag** [reply]
That's a precious info. I don't know why i didn't try to mention that. I will try that way. Thank you so much

**Comment by CliffAgius** [comment]
Closed as answered, if closed in error please let us know to re-open.
